sdk/framework/field_data.go

```diff
@@ -202,36 +202,12 @@ func (d *FieldData) getPrimitive(k string, schema *FieldSchema) (interface{}, bo
 		switch inp := raw.(type) {
 		case nil:
 			return nil, false, nil
-		case int:
-			result = inp
-		case int32:
-			result = int(inp)
-		case int64:
-			result = int(inp)
-		case uint:
-			result = int(inp)
-		case uint32:
-			result = int(inp)
-		case uint64:
-			result = int(inp)
-		case float32:
-			result = int(inp)
-		case float64:
-			result = int(inp)
-		case string:
+		default:
 			dur, err := parseutil.ParseDurationSecond(inp)
 			if err != nil {
 				return nil, false, err
 			}
 			result = int(dur.Seconds())
-		case json.Number:
```

Comment:
Does this case need handling in the updated `ParseDurationSecond`?

Reply:
ParseDurationSecond already handles it but differently -- it's above where the diff cuts out which is why you don't see it. It gets it as a string, then lets strconv.ParseInt do its thing (under the hood, json.Number.Int64 is doing exactly that anyways).